src/user-management/interfaces/authentication-events.interface.ts

```diff
@@ -0,0 +1,27 @@
+type EventStatus =
+  | {
+      status: 'succeeded';
+    }
+  | {
+      status: 'failed';
+      error: {
+        code: string;
+        message: string;
+      };
+    };
+
+export type EmailVerification = {
+  object: 'email_verification';
```

Comment:
we have a `type` field but no `object` for these since they don't correspond to an API object.